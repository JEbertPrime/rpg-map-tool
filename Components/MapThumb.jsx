import styled from "styled-components";
const Child = styled.div`
  display: none;
  align-items: center;
`;
const Parent = styled.div`
  &:hover ${Child} {
    display: block;
  }
  border-radius: 10px;
`;
const Wrapper = styled.div`
    margin:5px
`
const MapTitle = styled.h3`
    text-overflow:ellipsis;
`

export default function MapThumb({
  width,
  height,
  onClick,
  id,
  map,
  ...props
}) {
  var content = (
    <Wrapper>
      <MapTitle>{map.title}</MapTitle>
      <Parent
        id={id}
        onClick={onClick}
        style={{
          width: `${width}px`,
          height: `${height}px`,
          backgroundImage: `url(${map.file})`,
          backgroundSize: "cover",
        }}
      >
        <Child>{props.children}</Child>
      </Parent>
    </Wrapper>
  );

  return content;
}
