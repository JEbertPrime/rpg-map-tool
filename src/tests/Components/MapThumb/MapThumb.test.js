import { render, screen } from "@testing-library/react";
import MapThumb from "../../../../Components/MapThumb";
import renderer from "react-test-renderer";

const devMaps = [
  { title: "test_thumb", fileName: "maps/example_1.jpg", _id: 1 },
  { title: "test_thumb", fileName: "maps/example_2.jpg", _id: 2 },
];

describe("MapThumb", () => {
  it("renders without crashing", () => {
    render(<MapThumb map={devMaps[0]}></MapThumb>);
    expect(screen.getByText('test_thumb')).toBeInTheDocument();
  });
  it("renders child components", () => {
    const testText = "test_child";
    render(
      <MapThumb map={devMaps[0]} id={1}>
        <h1>{testText}</h1>
      </MapThumb>
    );

    expect(screen.getByText(testText)).toBeInTheDocument();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MapThumb map={devMaps[0]} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
