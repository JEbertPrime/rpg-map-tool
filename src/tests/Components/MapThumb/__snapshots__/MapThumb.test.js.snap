// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapThumb renders correctly 1`] = `
<div
  className="MapThumb__Wrapper-sc-4r9j8g-2 hpzgfY"
>
  <h3
    className="MapThumb__MapTitle-sc-4r9j8g-3 fHyouP"
  >
    test_thumb
  </h3>
  <div
    className="MapThumb__Parent-sc-4r9j8g-1 bUNUyg"
    style={
      Object {
        "backgroundImage": "url(maps/example_1.jpg)",
        "backgroundSize": "cover",
        "height": "undefinedpx",
        "width": "undefinedpx",
      }
    }
  >
    <div
      className="MapThumb__Child-sc-4r9j8g-0 dOGsgr"
    />
  </div>
</div>
`;
